### PR TITLE
Add spring security

### PR DESCRIPTION
**Pull request by RomanSoproniuk**

### CODE REVIEW

File: src/main/java/book/store/onlinebookstore/controller/BookController.java [hunk 1]
```diff
@@ -28,33 +29,38 @@
 public class BookController {
     private final BookService bookService;
 
+    @PreAuthorize("hasRole({'ROLE_USER', 'ROLE_ADMIN'})")
```
Comment (olekskov) [inline src/main/java/book/store/onlinebookstore/controller/BookController.java:32]: If all roles should have access to endpoint there is no sense to mention them all in PreAuthorize
You security property .anyRequest().authenticated() will require user to be logged-in
```suggestion
```
Please recheck all occurances

File: src/main/resources/db/changelog/changes/005-add-user-and-admin-role-to-roles-table.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+                  value: "ADMIN"
+        - insert:
+            tableName: roles
+            columns:
+              - column:
+                  name: id
+                  value: "2"
+              - column:
+                  name: role_name
+                  value: "USER"
```
Comment (olekskov) [inline src/main/resources/db/changelog/changes/005-add-user-and-admin-role-to-roles-table.yaml:23]: Please add empty lines into patch-sets where missing

**Review by olekskov — CHANGES_REQUESTED** [review]

File: src/main/java/book/store/onlinebookstore/controller/BookController.java [hunk 3]
```diff
@@ -28,33 +29,38 @@
 public class BookController {
     private final BookService bookService;
 
+    @PreAuthorize("hasRole('ROLE_USER')")
```
Comment (olekskov) [inline src/main/java/book/store/onlinebookstore/controller/BookController.java:32]: Are you sure admin shouldn't have access to get operations?

**Review by olekskov — APPROVED** [review]

File: src/main/java/book/store/onlinebookstore/controller/BookController.java (hunk 3 above)
Comment (RomanSoproniuk) [reply src/main/java/book/store/onlinebookstore/controller/BookController.java:32]: ok i will change it in next version)

**Pull request merged**